src/msix/msix.cpp

```diff
@@ -70,6 +70,26 @@ MSIX_API HRESULT STDMETHODCALLTYPE UnpackPackage(
     return static_cast<HRESULT>(MSIX::Error::OK);
 } CATCH_RETURN();
 
+MSIX_API HRESULT STDMETHODCALLTYPE UnpackPackageFromPackageReader(
+    IAppxPackageReader* packageReader,
+    MSIX_PACKUNPACK_OPTION packUnpackOptions,
+    MSIX_VALIDATION_OPTION validationOption,
+    char* utf8Destination) noexcept try
+{
+    ThrowErrorIfNot(MSIX::Error::InvalidParameter,
+        utf8Destination != nullptr,
+        "Invalid parameters"
+    );
+
+    auto to = MSIX::ComPtr<IStorageObject>::Make<MSIX::DirectoryObject>(utf8Destination);
+
+    MSIX::ComPtr<IAppxPackageReader> reader;

```

Comment:
>IAppxPackageReader [](start = 17, length = 18)

You are doing an extra QI
1- QI to IAppxPackageReader
2 -QI to IPackageReader in ComPtr::As

You can make it work with only one QI